Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AddAnnex/BusinessRules/AnnexCanOnlyBeAddedOnlyBeAddedToActiveBindingContractRule.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace EvolutionaryArchitecture.Fitnet.Contracts.Core.AddAnex.BusinessRules;
+
+using Common.Core.BusinessRules;
+
+internal sealed class AnnexCanOnlyBeAddedOnlyBeAddedToActiveBindingContractRule : IBusinessRule
+{
+    private readonly DateTimeOffset? _terminatedAt;
+    private readonly DateTimeOffset _expiringAt;
+
+    internal AnnexCanOnlyBeAddedOnlyBeAddedToActiveBindingContractRule(
+        DateTimeOffset? terminatedAt,
+        DateTimeOffset expiringAt)
+    {
+        _terminatedAt = terminatedAt;
+        _expiringAt = expiringAt;
+    }
+
+    public bool IsMet() => !_terminatedAt.HasValue && _expiringAt > DateTimeOffset.UtcNow;
```

Comment:
Using DateTimeOffset.UtcNow here force to use DateTime.Now in unit tests. When the test is executed, result can be non determinictic i would just pass ISystemClock as paramter to bussiness rule and mock it in test with static date. What you think?

Reply:
But then we should use SystemClock in every single place in our code - we have a lot of different places where we are mixing DateTimeOffset.Now with DateTimeOffset.UtcNow.

Additionally, passing ISystemClock to business rule would require to pass the instance of SystemClock from the entity itself (we manually create the business rule). 

What do you think to handle it outside of this PR in a separate branch and change it everywhere in chapter 4?

Reply:
Yeah, the time provider should be already available in our project. Let's use it